mlflow/models/model.py

```diff
@@ -922,10 +922,15 @@ def set_model(model):
 
     if not isinstance(model, PythonModel):
         try:
-            from mlflow.langchain import _validate_and_wrap_lc_model
+            from mlflow.langchain import _validate_and_prepare_lc_model_or_path
 
-            # If its not a PyFuncModel, then it should be a Langchain model
-            _validate_and_wrap_lc_model(model, None)
+            # If its not a PyFuncModel, then it should be a Langchain model (not a path)
+            # Check this since the validation function does not
+            if isinstance(model, str):
+                raise mlflow.MlflowException(
+                    "Model should either be an instance of PyFuncModel or Langchain type."
+                )
+            model = _validate_and_prepare_lc_model_or_path(model, None)
```

Comment:
@annzhang-db @sunishsheth2009 I recall that we talked about making `set_model` support a Python function too. This is also supported by `mlflow.pyfunc.log_model()`. Can we support this if it's easy or file a follow-up internal ticket?

Reply:
Yes we have a ticket for that option as well. We will have to do it as a follow up. :)